partner_ref_unique/tests/test_res_partner_ref.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2017 Tecnativa - Vicent Cubells
+# Copyright 2020 Tecnativa - João Marques
+# Copyright 2020 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.exceptions import ValidationError
+from odoo.tests import common
+
+
+class TestResPartnerRefUnique(common.SavepointCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestResPartnerRefUnique, cls).setUpClass()
+        cls.partner_obj = cls.env["res.partner"]
+        cls.company_obj = cls.env["res.company"]
+        # look for possible already duplicated refs for being resilient
+        cls.partner_obj.search([("ref", "!=", False)]).write({"ref": False})
+        cls.company = cls.company_obj.create({"name": "Test company"})
+        cls.env.user.write(
+            {"company_id": cls.company.id, "company_ids": cls.company.ids}
+        )
+        cls.partner1 = cls.partner_obj.create({"name": "Partner1", "company_id": False})
+        cls.partner2 = cls.partner_obj.create({"name": "Partner2", "company_id": False})
+
+    def test_check_ref_company(self):
+        (self.partner1 + self.partner2).write({"company_id": self.company.id})
+        # Test that we can create/modify partners with same ref in current situation
+        self.partner1.ref = "same_ref"
+        partner = self.partner_obj.create({"name": "other", "ref": "same_ref"})
+        # Try to activate restriction
+        with self.assertRaises(ValidationError):
```

Comment:
```suggestion
        with self.assertRaisesRegex(ValidationError, "This reference is equal to partner"):
```
it would be a good idea to change all assertRaises to assertRaisesRegex

Reply:
why?

Reply:
This way we ensure that the correct error is raised.

Reply:
updated in latest commit